### DISCUSSION TITLE
When creating a new mesh, the bl-touch goes off the platform

### DISCUSSION THREAD

**Discussion by KPrinz** [Jyers/Marlin#237] [Q&A] [answered]
So, I just installed a BL-Touch to an Ender 3 v2. As the probe sits about 41mm to the left, how do I account for that? When I go on auto bed levelling, the probe goes off the left end and probes the air, while the nozzle touches the glass....

**Comment by Schnabulation** [comment, marked as answer]
You have to set the probe offset in Control --> Advanced.

**Reply by KPrinz** [reply]
Thanks! That worked like a charm! Can I leave it at the auto levelling, or do I still need to find a Z-offset with a sheet of paper? -> I do: it was something like 2.98. Printing now...

**Reply by Schnabulation** [reply]
> I do: it was something like 2.98. Printing now...

Based on this I am 99% sure you are using the official Creality BL Touch mount. ;-) I have the same Z-probe offset...

**Reply by KPrinz** [reply]
Exactly, the one that came with the v3.1 set

**Reply by KPrinz** [reply]
> > I do: it was something like 2.98. Printing now...
> 
> Based on this I am 99% sure you are using the official Creality BL Touch mount. ;-) I have the same Z-probe offset...

Can you please share your x and y offsets for that setup? I am lacking the precision measuring tool atm...

**Reply by Schnabulation** [reply]
Of course: I‘m at -45 and -7.

**Reply by KPrinz** [reply]
Merci vielmals!

**Reply by Schnabulation** [reply]
Gerne! Wenn du Fragen hast, melde dich ;-)

**Reply by danrough** [reply]
@KPrinz @Schnabulation could I ask you both to help me understand this a little please?

The -45 and -7 numbers that are being quoted above see my BLTouch probe (also bought in the official antclabs 3.1 kit, though my Z Offset is -4.32) miles over the edge on any mesh X Axis points less than 3, and hitting the end stop on Y Axis points greater than 11.

Is there an official way to calculate these values, or are they trial and error?

If I use the defaults of -10 and -10, then the mesh X Axis point number 2 barely touches the edge of the bed, whereas the Y Axis mesh points all seem accessible (i.e. it would avoid the standard Ender metal clip).

Is there a guide that either of you used that you'd be prepared to share? I'd really like to understand how people are arriving at these values, so that I can work them out for myself.

**Reply by KPrinz** [reply]
> Is there a guide that either of you used that you'd be prepared to share? I'd really like to understand how people are arriving at these values, so that I can work them out for myself.

The X and Y offset describe the difference in position between the tip of the BLTouch probe and the extruder nozzle. Best measured with a caliper.

**Reply by Schnabulation** [reply]
> The -45 and -7 numbers that are being quoted above see my BLTouch probe (also bought in the official antclabs 3.1 kit, though my Z Offset is -4.32) miles over the edge on any mesh X Axis points less than 3, and hitting the end stop on Y Axis points greater than 11.

Are you sure you are using -45 for X and -7 for Y and not the other way around? Also you are sure you are using negatives?

**Reply by danrough** [reply]
![IMG_0570](https://user-images.githubusercontent.com/3875399/111980327-0b3ff800-8afe-11eb-8f75-c7b80c4b674b.jpeg)

These are the values that I am using. After setting them, I used the menu functionality to store the settings. Subsequent changes to them - back to the defaults of -10, -10 for example, produce different results, and so I am working on the basis that the method that I am using to set these values is correct.

With these values, the Y Axis seems to be fine. I was a little unsure about whether the probe will clear the standard Creality metal clips, and so I decreased my Y value from -7 to -6. I will do more checks to fine tune this, when I have the X Axis issues sorted.

The X axis though is definitely off, unless the probe does not travel to mesh points less than 3 on the X axis. I used a vernier caliper to measure the distance between the probe and the nozzle on my printer and I came up with ~41.7mm and so the value of 42 is definitely within a tolerance I can live with for the purpose of this investigation. 

I looks like there's quite a few people who've raised this issue. I'm going to go through all the topics that they have created, to see if I have missed something along the way.

**Reply by Schnabulation** [reply]
> I was a little unsure about whether the probe will clear the standard Creality metal clips, and so I decreased my Y value from -7 to -6.

Please don't do that! The offset needs to be exact because it's how the printer estimates the position of the probe relative to the nozzle. If you change that it will mess with your mesh.

If you are worried about the probe hitting the metal clips then look at `probing_margins` in the config-files.

**Reply by danrough** [reply]
>If you are worried about the probe hitting the metal clips then look at probing_margins in the config-files.

Good to know. Thank you :)
